kludges.js

```diff
@@ -388,16 +202,22 @@ function restoreMethod(obj, name) {
  * move from gnome version to gnome version.  Next to the swipe tracker locations
  * below are the gnome versions when they were first (or last) seen.
  */
-var swipeTrackers = [
-    Main?.overview?._swipeTracker, // gnome 40+
-    Main?.overview?._overview?._controls?._workspacesDisplay?._swipeTracker, // gnome 40+
-    Main?.wm?._workspaceAnimation?._swipeTracker, // gnome 40+
-    Main?.wm?._swipeTracker // gnome 38 (and below)
-].filter(t => typeof t !== 'undefined');
-
 var signals;
-function init() {
-    registerOverridePrototype(imports.ui.messageTray.MessageTray, '_updateState');
+var swipeTrackers;
+var settings;
+var wmSettings;
+function startup() {
+    settings = ExtensionUtils.getSettings();
+    wmSettings = new Gio.Settings({schema_id: 'org.gnome.desktop.wm.preferences'});
```

Comment:
Unfortunately I can't attach a comment to an unchanged line... So here everything in one comment. I would:

- Put all `registerOverridePrototype` calls into `enableOverride` (and maybe rename it to `setupOverrides`.
- Put all the swipe tracker related stuff in a function `setupSwipeTrackers`
- Put these lines in `enable` and also the line `signals = new utils.Signals();`

And we could similarly split up the `enable` function. E.g. into:

- `setupSignals`
- `monkeyPatchWindowManager`
- `monkeyPatchWorkspace`
- `monkeyPatchMessageTray`

(I don't know what modifying the prototype in JS is usually called, but "monkey patching" is often used in Python for similar things)

Reply:
Hey @Lythenas!  I've implemented these (check the latest).  Simplified quite a bit and implemented the override functions into one place.

Also removed MessageTray as not needed/used now.

Cheers!